qoqo_for_braket/qoqo_for_braket/queued_results.py

```diff
@@ -155,3 +156,132 @@ def poll_result(
             else:
                 return None
         return self._results
+
+
+class QueuedProgramRun:
+    """Queued Result of the measurement."""
+
+    def __init__(self, measurement: Any, queued_circuits: List[QueuedCircuitRun]) -> None:
+        """Initialise the QueuedProgramRun class.
+
+        Args:
+            measurement: the qoqo Measurement to run
+            queued_circuits: the list of associated queued circuits
+        """
+        self._measurement = measurement
+        self._queued_circuits: List[QueuedCircuitRun] = queued_circuits
+        self._registers: Tuple[
+            Dict[str, List[List[bool]]],
+            Dict[str, List[List[float]]],
+            Dict[str, List[List[complex]]],
+        ] = ({}, {}, {})
+        for circuit in self._queued_circuits:
+            if circuit._results is not None:
+                self._registers[0].update(circuit._results[0])
+                self._registers[1].update(circuit._results[1])
+                self._registers[2].update(circuit._results[2])
+
+    def poll_result(
+        self,
+    ) -> Optional[
+        Tuple[
+            Dict[str, List[List[bool]]],
+            Dict[str, List[List[float]]],
+            Dict[str, List[List[complex]]],
+        ]
+    ]:
+        """Poll the result once.
+
+        Returns:
+            Optional[Tuple[Dict[str, List[List[bool]]],
+                           Dict[str, List[List[float]]],
+                           Dict[str, List[List[complex]]],]
+                     ]: Result if all tasks were successful.
+
+        Raises:
+            RuntimeError: job failed or cancelled
+        """
+        all_finished = [False] * len(self._queued_circuits)
+        for i, queued_circuit in enumerate(self._queued_circuits):
+            res = queued_circuit.poll_result()
+            if res is not None:
+                self._registers[0].update(res[0])  # add results to bit registers
+                self._registers[1].update(res[1])  # add results to float registers
+                self._registers[2].update(res[2])  # add results to complex registers
+                all_finished[i] = True
+
+        if not all(all_finished):
+            return None
+        else:
+            return self._registers
```

Comment:
Here we can check if the stored measurement is a ClassicalRegister and in that case just return the registers and if it's another type of measurement just call .evaluate.

Reply:
Modified